WordPress/Classes/Services/BlogService.m

```diff
@@ -244,6 +249,60 @@ - (void)syncBlog:(Blog *)blog success:(void (^)())success failure:(void (^)(NSEr
     }
 }
 
+- (void)batchSyncBlog:(Blog *)blog
+{
+    __weak __typeof(self) weakSelf = self;
+
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:self.managedObjectContext];
+
+    [postService syncPostsOfType:PostServiceTypePage forBlog:blog success:^{
+
+        [postService syncPostsOfType:PostServiceTypePost forBlog:blog success:^{
+
+            id<BlogServiceRemote> remote = [weakSelf remoteForBlog:blog];
```

Comment:
Shouldn't the `weakSelf` go strong here? e.g. `__typeof (self) strongSelf = weakSelf;`